Undefined names: 'x' and 'rst'

[flake8](http://flake8.pycqa.org) testing of https://github.com/SmirkCao/Lihang on Python 3.7.1

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./CH11/crf.py:60:59: F821 undefined name 'x'
                    self.M[i, j, :] = self.f[k](y_i, y_j, x, j)
                                                          ^
./CH10/hmm.py:153:24: F821 undefined name 'rst'
                return rst
                       ^
2     F821 undefined name 'rst'
2
```
__E901,E999,F821,F822,F823__ are the "_showstopper_" [flake8](http://flake8.pycqa.org) issues that can halt the runtime with a SyntaxError, NameError, etc. These 5 are different from most other flake8 issues which are merely "style violations" -- useful for readability but they do not effect runtime safety.
* F821: undefined name `name`
* F822: undefined name `name` in `__all__`
* F823: local variable name referenced before assignment
* E901: SyntaxError or IndentationError
* E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree

--- a/CH10/hmm.py
+++ b/CH10/hmm.py
@@ -149,8 +149,8 @@ class HMM(object):
             self._do_estep(X)
             self._do_mstep(X)
             # convergence check
-            if False:
-                return rst
+        #    if False:
+        #        return rst
         return self
 
     def decode(self, X):

--- a/CH11/crf.py
+++ b/CH11/crf.py
@@ -57,7 +57,7 @@ class CRF(object):
         for k in range(self.K):
             for i, y_i in enumerate(self.y):
                 for j, y_j in enumerate(self.y):
-                    self.M[i, j, :] = self.f[k](y_i, y_j, x, j)
+                    self.M[i, j, :] = self.f[k](y_i, y_j, i, j)
 
         return self
 

